src/components/backend-ai-credential-list.ts

```diff
@@ -696,25 +697,91 @@ export default class BackendAICredentialList extends BackendAIPage {
       `, root);
   }
 
+  _validateRateLimit() {
+    // this._adjustRateLimit();
+    const warningRateLimit = 100;
+    const maximumRateLimit = 50000; // the maximum value of rate limit value
+
+    this.rateLimit.validityTransform = (newValue, nativeValidity) => {
+      if (!nativeValidity.valid) {
+        if (nativeValidity.valueMissing) {
+          this.rateLimit.validationMessage = _text('credential.RateLimitInputRequired');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        } else if (nativeValidity.rangeOverflow) {
+          this.rateLimit.value = newValue = maximumRateLimit.toString();
+          this.rateLimit.validationMessage = _text('credential.RateLimitValidation');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        } else if (nativeValidity.rangeUnderflow) {
+          this.rateLimit.value = newValue = '1';
+          this.rateLimit.validationMessage = _text('credential.RateLimitValidation');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        } else {
+          this.rateLimit.validationMessage = _text('credential.InvalidRateLimitValue');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        }
+      } else {
+        if (newValue.length !== 0 && !isNaN(Number(newValue)) && Number(newValue) < warningRateLimit) {
+          this.rateLimit.validationMessage = _text('credential.WarningLessRateLimit');
+          return {
+            valid: !nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        }
+        return {
+          valid: nativeValidity.valid,
+          customError: !nativeValidity.valid
+        };
+      }
+    };
+  }
+
+  openDialog(id) {
+    (this.shadowRoot?.querySelector('#' + id) as BackendAIDialog).show();
+  }
+
+  closeDialog(id) {
+    (this.shadowRoot?.querySelector('#' + id) as BackendAIDialog).hide();
+  }
+
   /**
    * Save a keypair modification.
    *
-   * @param {Event} e - Dispatches from the native input event each time the input changes.
+   * @param {boolean} confirm - Save keypair info even if rateLimit is less the warningRateLimit if `confirm` is true.
```

Comment:
When setting `closeWithConfirm` to `keypair-modify-dialog`, the `keypair-modify-dialog` cannot be closed immediately by pressing the close button. So I added confirm param to this function.

Reply:
I'll check it.

Reply:
[backend-ai-credential-list.ts.zip](https://github.com/lablup/backend.ai-webui/files/9650443/backend-ai-credential-list.ts.zip)
this is the version of adding confirm dialog